openscm/utils.py

```diff
@@ -0,0 +1,33 @@
+"""Utility functions for openscm.
+"""
+
+import warnings
+
+
+import functools
+
+
+def counted(base_func):
+    """Count how many times a function has been called
+    """
+    # a good idea according to
+    # https://realpython.com/primer-on-python-decorators/#simple-decorators...
+    @functools.wraps(base_func)
+    def wrapped(*args, **kwargs):
+        wrapped.calls += 1
+
+        return base_func(*args, **kwargs)
+
+    wrapped.calls = 0
+
+    return wrapped
+
+
+@counted
+def ensure_input_is_tuple(inp):
+    if isinstance(inp, str):
+        if ensure_input_is_tuple.calls == 1:
```

Comment:
Do we really need a decorator here (that itself uses a decorated function)? The more straight forward way would be:
```suggestion
        if ensure_input_is_tuple.get_attr('calls', 0) == 1:
            ensure_input_is_tuple.calls = 1
```